src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -0,0 +1,90 @@
+        {
+            Debug.Assert(startupHooksVariable != IntPtr.Zero);
+            string[] startupHooks = Marshal.PtrToStringUTF8(startupHooksVariable).Split(new char[] { Path.PathSeparator });
+
+            // Parse startup hook variable
+            List<Tuple<string, string>> startupHooksParsed = new List<Tuple<string, string>>();
+            foreach (string startupHook in startupHooks)
+            {
+                string[] assemblyPathAndTypeName = startupHook.Split(new char[] { '!' });
+                if (assemblyPathAndTypeName.Length != 2)
+                {
+                    throw new ArgumentException(SR.Argument_InvalidStartupHookSyntax);
+                }
+                string assemblyPath = assemblyPathAndTypeName[0];
+                string typeName = assemblyPathAndTypeName[1];
+                if (String.IsNullOrEmpty(assemblyPath) || String.IsNullOrEmpty(typeName))
+                {
+                    throw new ArgumentException(SR.Argument_InvalidStartupHookSyntax);
+                }
+                startupHooksParsed.Add(Tuple.Create(assemblyPath, typeName));
+            }
+
+            // Ensure the startup dlls exist
+            foreach (var startupHook in startupHooksParsed)
+            {
+                var assemblyPath = startupHook.Item1;
+                if (!File.Exists(assemblyPath))
+                {
+                    throw new FileNotFoundException(SR.Format(SR.FileNotFound_ResolveAssembly, assemblyPath));
+                }
+            }
+
+            // Call each hook in turn
+            foreach (var startupHook in startupHooksParsed)
+            {
+                CallStartupHook(startupHook.Item1, startupHook.Item2);
+            }
+        }
+
+        // Load the specified assembly, and call the specified type's
+        // "public static void Initialize()" method.
+        private static void CallStartupHook(string assemblyPath, string typeName)
+        {
+            Debug.Assert(!String.IsNullOrEmpty(assemblyPath));
+            Debug.Assert(!String.IsNullOrEmpty(typeName));
+            Debug.Assert(File.Exists(assemblyPath));
+
+            string fullAssemblyPath = Path.GetFullPath(assemblyPath);
+            Assembly assembly = AssemblyLoadContext.Default.LoadFromAssemblyPath(fullAssemblyPath);
+            Debug.Assert(assembly != null);
+            Type type = assembly.GetType(typeName, throwOnError: true);
+            MethodInfo initializeMethod = type.GetMethod(initializeMethodName, Type.EmptyTypes);
+
+            if (initializeMethod == null)
+            {
+                throw new MissingMethodException(typeName, initializeMethodName);
+            }
+
+            Debug.Assert(initializeMethod.IsPublic);
+            if (!(initializeMethod.IsStatic && initializeMethod.ReturnType == typeof(void)))
+            {
+                throw new ArgumentException(SR.Format(SR.Argument_InvalidStartupHookSignature, typeName + "." + initializeMethodName));
```

Comment:
[`Type.Delimiter`](https://docs.microsoft.com/en-us/dotnet/api/system.type.delimiter?view=netcore-2.1)

Reply:
Will use this instead, thanks!